test/Microsoft.AspNet.Mvc.Core.Test/OutputFormatterTests.cs

```diff
@@ -76,24 +76,31 @@ public static IEnumerable<object[]> SelectResponseCharacterEncodingData
         }
 
         [Fact]
-        public void WriteResponseContentHeaders_FormatterWithNoEncoding_Throws()
+        public void WriteResponseContentHeaders_NoSupportedEncodings_NoEncodingIsSet()
```

Comment:
Does this one test cover the original poster's scenario and related scenarios well enough? E.g. seems to cover only the "we assume this is a non-text format" case and not the "custom implementation of SelectCharacterEncoding" case.